### PR TITLE
US887031: Use secure Trust Manager when connecting to RabbitMQ via TLS

### PR DESCRIPTION
**Pull request by jonny-mccoubrey**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=887031

### CODE REVIEW

**Comment by jonny-mccoubrey** [comment]
According to `rabbitmq.conf` file, we are using TLS v1.2

**Comment by jonny-mccoubrey** [comment]
New exceptions for consumers to handle so therefore breaking changes

File: worker-test/src/main/docker/rabbitmq/certs/RMQ-CA-cert.pem [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+YmeB5sXFC/SImEVsVRr3vEAg0nxQCCoYjQIDAQABo1MwUTAdBgNVHQ4EFgQUOMUp
+xYtx5AhhswWMTe7Tza7iWXEwHwYDVR0jBBgwFoAUOMUpxYtx5AhhswWMTe7Tza7i
+WXEwDwYDVR0TAQH/BAUwAwEB/zANBgkqhkiG9w0BAQsFAAOCAQEAQQkORj6Ie/1i
+gTj5D7WBVWi0nr74UI3xNqN+UV8QnqVW+bITbmdr1/kIlzlHLUIyWvgVK9BWUmgo
+LFynQgv0LDpQSpbvt8/01YqdqRHxCSKxNTz0nX7ZHOM3hSM8htRdrS7g8eFOmpsO
+UCOfnq0ajbk5U2qu0FdMLerfwqVoT3wW0UtLrOfiGNKWuIweVPG1MqPUAVRqQ5Aj
+w8TbgjrTQiazahABCTApJoVc5PVJt59S3Iz/oI85il8XMCs7ohllj2VyWubk4G25
+TgBzeNfYr0xTM9bLYsVARe2y/WxF2Qt1+XfTIp42XCLKHGwqUvgcas9cjC9Tsh8A
+scXkgT9gPg==
+-----END CERTIFICATE-----
```
Comment (dermot-hardy) [inline worker-test/src/main/docker/rabbitmq/certs/RMQ-CA-cert.pem]: Why are we checking in certificates?  Can't we generate these during the build?

Comment (jonny-mccoubrey) [reply worker-test/src/main/docker/rabbitmq/certs/RMQ-CA-cert.pem]: I was just trying this on first pass but spoke to Andy about spinning up a `keystore` image to generate certificates. Just hadn't checked it in until this morning.

File: worker-queue-rabbit/src/main/java/com/hpe/caf/worker/queue/rabbit/RabbitWorkerQueue.java [hunk 2]
```diff
@@ -324,8 +325,8 @@ public HealthResult healthCheck()
     }
 
     private void createConnection(TaskCallback callback, WorkerConfirmListener listener)
-        throws IOException, TimeoutException, URISyntaxException, NoSuchAlgorithmException, KeyManagementException
-    {
+            throws IOException, TimeoutException, URISyntaxException, NoSuchAlgorithmException,
```
Comment (andyreidz) [inline worker-queue-rabbit/src/main/java/com/hpe/caf/worker/queue/rabbit/RabbitWorkerQueue.java]:
```suggestion
        throws IOException, TimeoutException, URISyntaxException, NoSuchAlgorithmException,
```

File: util-rabbitmq/src/main/java/com/hpe/caf/util/rabbitmq/RabbitUtil.java [hunk 3]
```diff
@@ -60,8 +65,8 @@ private RabbitUtil()
      * @throws TimeoutException if the connection fails to establish
      */
     public static Connection createRabbitConnection(String protocol, String host, int port, String user, String pass)
-        throws IOException, TimeoutException, URISyntaxException, NoSuchAlgorithmException, KeyManagementException
-    {
+        throws IOException, TimeoutException, URISyntaxException, NoSuchAlgorithmException,
+            KeyManagementException, KeyStoreException {
```
Comment (andyreidz) [inline util-rabbitmq/src/main/java/com/hpe/caf/util/rabbitmq/RabbitUtil.java]: Let's do this how you showed me jersey does it and and catch KeyStoreException, throwing an IllegalStateException and avoiding another breaking change in the framework.

https://github.com/eclipse-ee4j/jersey/blob/85017aaee97d8a9f1b2dcdab7d3da5c0e57130e4/core-common/src/main/java/org/glassfish/jersey/SslConfigurator.java#L764

File: worker-test/pom.xml [hunk 4]
```diff
@@ -170,6 +170,25 @@
...
+                                    <runCmd>openssl req -x509 -new -key /test-keystore/ca_key.pem -out /test-keystore/ca_certificate.pem -days 3650 -subj "/CN=myname/OU=myorganisational.unit/O=myorganisation/L=mycity/S=myprovince/C=GB"</runCmd>
+                                    <runCmd>openssl genrsa -out /test-keystore/server_key.pem 2048</runCmd>
+                                    <runCmd>openssl req -new -key /test-keystore/server_key.pem -out /test-keystore/server.csr -subj "/CN=myname/OU=myorganisational.unit/O=myorganisation/L=mycity/S=myprovince/C=GB"</runCmd>
+                                    <runCmd>chmod 664 /test-keystore/server_key.pem</runCmd>
+                                    <runCmd>openssl x509 -req -in /test-keystore/server.csr -CA /test-keystore/ca_certificate.pem -CAkey /test-keystore/ca_key.pem -CAcreateserial -out /test-keystore/server_certificate.pem -days 3650</runCmd>
+                                </runCmds>
+                                <volumes>
+                                    <volume>/test-keystore</volume>
+                                </volumes>
+                            </build>
```
Comment (andyreidz) [inline worker-test/pom.xml:190]: Need a run section for arm64 systems that run the images under amd64 emulation
```suggestion
                            </build>
                            <run>
                                <platform>linux/amd64</platform>
                            </run>
```

**Review by andyreidz — CHANGES_REQUESTED** [review]
Looking good, some minor feedback

**Review by andyreidz — APPROVED** [review]
Good stuff, thanks!

**Pull request merged**